### PR TITLE
HSTS patch 1

### PR DESCRIPTION
**Pull request by garethsb**
Tweaks only

### CODE REVIEW

**Comment by garethsb** [comment]
Not 100% certain that turning HSTS header on/off based on `client_secure` is correct... depends if we running behind a secure reverse proxy, we should be responsible for adding the header or the proxy should be?

**Comment by garethsb** [comment]
Commit 86eabe092a3539ecf305c54632a503f606367851 makes the change to use `bst::optional`, not sure whether it's better or not? 🤷

(The ternary `?:` repeated in _nmos/node_server.cpp_ and _nmos/registry_server.cpp_ could be factored into a function in _nmos/settings.h_ though then `#include "bst/optional.h"` would be needed there. 😕)

File: Development/cpprest/http_utils.cpp [hunk 1]
```diff
@@ -466,27 +466,32 @@ namespace web
...
+            }
+
             hsts parse_hsts_header(const utility::string_t& value)
             {
                 hsts result;
                 auto directives = parse_directives_header(value);
-                const auto max_age = std::find_if(directives.begin(), directives.end(), [](const directive& directive) { return boost::algorithm::iequals(directive.first, U("max-age")); });
+                // hm, required
+                const auto max_age = find_directive(directives, U("max-age"));
                 if (directives.end() != max_age) result.max_age = utility::istringstreamed(max_age->second, 0);
```
[the remark this answers is not in the document]
Comment (garethsb) [reply Development/cpprest/http_utils.cpp]: Have added the check for missing max-age, and a comment that max-age=meow is treated as max-age=0, and lots more unit tests.

**Comment by garethsb** [comment]
Agreed with @lo-simon that the `bst::optional` version was easier to maintain, so I've done some clean up and force pushed. I then found a bug in the handling of quoted strings in directive values, so I've fixed that and added more unit tests. Assuming this builds successfully on all platforms in CI, this is good for a final review...

**Review by lo-simon — APPROVED** [review]
LGTM

**Pull request merged**